File: clients/sellingpartner-api-aa-csharp/src/Amazon.SellingPartnerAPIAA/LWAAuthorizationCredentials.cs
using System;
using System.Collections.Generic;

namespace Amazon.SellingPartnerAPIAA
{
    public class LWAAuthorizationCredentials
    {
        public LWAAuthorizationCredentials()
        {
            this.Scopes = new List<string>();
        }

        /**
         * LWA Client Id
         */
        public string ClientId { get; set; }

        /**
         * LWA Client Secret
         */
        public string ClientSecret { get; set; }

        /**
         * LWA Refresh Token
         */
        public string RefreshToken { get; set; }

        /**
         * LWA Authorization Server Endpoint
         */
        public Uri Endpoint { get; set; }

        /**
         * LWA Authorization Scopes
         */
        public List<string> Scopes { get; set; }
    }
}

File: clients/sellingpartner-api-aa-java/src/com/amazon/SellingPartnerAPIAA/LWAAuthorizationCredentials.java
package com.amazon.SellingPartnerAPIAA;

import lombok.Builder;
import lombok.Data;
import lombok.NonNull;

import java.util.Arrays;
import java.util.HashSet;

/**
 * LWAAuthorizationCredentials
 */
@Data
@Builder
public class LWAAuthorizationCredentials {
    /**
     * LWA Client Id
     */
    @NonNull
    private String clientId;

    /**
     * LWA Client Secret
     */
    @NonNull
    private String clientSecret;

    /**
     * LWA Refresh Token
     */
    private String refreshToken;

    /**
     * LWA Authorization Server Endpoint
     */
    @NonNull
    private String endpoint;

    /**
     * LWA Client Scopes
     */
    private LWAClientScopes scopes;

    public static class LWAAuthorizationCredentialsBuilder {

        {
            scopes = new LWAClientScopes(new HashSet<>());
        }

        public LWAAuthorizationCredentialsBuilder withScope(String scope) {
            return withScopes(scope);
        }

        public LWAAuthorizationCredentialsBuilder withScopes(String... scopes) {
            if (scopes != null) {
                Arrays.stream(scopes)
                        .forEach(this.scopes::addScope);
            }
            return this;
        }


    }

}
